core/input/input.cpp

```diff
@@ -1735,11 +1783,20 @@ void Input::set_fallback_mapping(const String &p_guid) {
 	}
 }
 
+bool Input::is_mapping_known(const StringName &p_guid) {
+	for (const JoyDeviceMapping &map : map_db) {
+		if (map.uid == p_guid) {
+			return true;
+		}
+	}
+	return false;
+}
+
 //platforms that use the remapping system can override and call to these ones
 bool Input::is_joy_known(int p_device) {
 	if (joy_names.has(p_device)) {
 		int mapping = joy_names[p_device].mapping;
-		if (mapping != -1 && mapping != fallback_mapping) {
+		if (mapping != -1 && mapping != fallback_mapping && !map_db[mapping].name.ends_with(" [auto]")) {
```

Comment:
`auto` should be flag in the `JoyDeviceMapping`, using name is not reliable and there's no reason to show it in the user readable name (a new getter can be added if it's necessary to access this information from the scrips).

Reply:
Now added an `auto_generated` flag for `JoyDeviceMapping`.